sample-client-app/sample-app-with-shared-lib/src/main/java/com/vmware/vipclient/sample/SharedLibDemo.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2019-2020 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vipclient.sample;
+
+import java.util.Locale;
+
+import com.vmware.vipclient.i18n.I18nFactory;
+import com.vmware.vipclient.i18n.VIPCfg;
+import com.vmware.vipclient.i18n.base.cache.FormattingCache;
+import com.vmware.vipclient.i18n.base.cache.MessageCache;
+import com.vmware.vipclient.i18n.base.instances.TranslationMessage;
+import com.vmware.vipclient.i18n.exceptions.VIPClientInitException;
+import com.vmware.vipclient.sample.shared.Shared;
+
+public class SharedLibDemo {
```

Comment:
This class is for showing how write code in shared library, so shouldn't this class put into sample-shared-library? 

Besides if it is not put into shared library, each application using shared library need write these code snippet.

Reply:
This is for showing how to use a shared library's class and method (Shared.someMethod in lines 33-34) in a root application code such as SharedLibDemo.

Each root application that uses the shared library will call shared.someMethod() as usual (line 34). Only those root applications that also want to use VIP will have to write lines 19-33.

Reply:
My description maybe not precise, but the precondition I based on is both the root application and shared library use VIP, or there is no need for us to write these demo code.

The goal I want to reach is also let the root application call the shared library code as usual(before use VIP), and no big change for shared library.

But if put below code snippet in the root application, and root application call many methods of shared library, you need pass 'tmSharedLib' to each constructor of shared lib class called by the root application as line 31~33,
       TranslationMessage tmSharedLib = (TranslationMessage)  I18nFactory.getInstance().getMessageInstance(TranslationMessage.class, sharedLibCfg);  
        Shared shared = new Shared(tmSharedLib, locale);
        System.out.println("Shared library's message; fetch triggered from inside the shared library: " + shared.someMethod());

 this not only need add constructor in shared library, but also not call shared lib methods as usual, because before use VIP it call shared lib may like this:
        Shared shared = new Shared();
        System.out.println("Shared library's message; fetch triggered from inside the shared library: " + shared.someMethod());

But if you move these codes into sample-shared-library,  no need add constructor in shared lib and can call share lib methods as usual.

Reply:
Yes, this sample is for the current implementation, which I did not design or implement. I am merely adding a sample usage through this PR. Improving the implementation can be in another PR.

Reply:
Actually, Shared.java may act as a singleton instance that holds tmSharedLib which is to be used for the rest of the shared library classes. The point is to demonstrate how to pass the tmSharedLib, not how to design the shared lib.

Reply:
Yeah, Shared.java may act as a singleton instance that holds tmSharedLib which is to be used for the rest of the shared library classes. good idea!

But in this case Shared class should be called by shared lib's methods which need call getMessage to get translation, not by the root application, because the methods root application need call must not all in the Shared.java.

I think we should design sample app to make it easiest to use and as close to the real application as possible. So suggest future improvement.

Reply:
@huihuiw01 , https://github.com/vmware/singleton/issues/744